### PR TITLE
[FLINK-16081][docs] Translate /dev/table/index.zh.md

### PR DESCRIPTION
**Pull request by libenchao**
## What is the purpose of the change

Translate /dev/table/streaming/index.zh.md

## Brief change log

Translate /dev/table/streaming/index.zh.md

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by wuchong** [comment]
Hi @JingsongLi , could you help to review this?

File: docs/dev/table/index.zh.md [hunk 1]
```diff
@@ -25,41 +25,41 @@ specific language governing permissions and limitations
...
-Dependency Structure
+依赖图
 --------------------
 
-Starting from Flink 1.9, Flink provides two different planner implementations for evaluating Table & SQL API programs: the Blink planner and the old planner that was available before Flink 1.9. Planners are responsible for
-translating relational operators into an executable, optimized Flink job. Both of the planners come with different optimization rules and runtime classes.
-They may also differ in the set of supported features.
+从1.9开始，Flink 提供了两个 table planner 实现来执行 Table API 和 SQL 程序：Blink planner 和 old planner，old planner 在1.9之前就已经存在了。
+planner 的作用主要是把关系型的操作翻译成可执行的、经过优化的 Flink job。这两个 planner 所使用的优化规则以及运行时都不一样。
+它们在支持的功能上也有些诧异。
```
Comment (JingsongLi) [inline docs/dev/table/index.zh.md]: 差异

File: docs/dev/table/index.zh.md [hunk 2]
```diff
@@ -25,41 +25,41 @@ specific language governing permissions and limitations
...
-* `flink-table-planner`: The table program planner and runtime. This was the only planner of Flink before the 1.9 release. It is still the recommended one.
-* `flink-table-planner-blink`: The new Blink planner.
-* `flink-table-runtime-blink`: The new Blink runtime.
-* `flink-table-uber`: Packages the API modules above plus the old planner into a distribution for most Table & SQL API use cases. The uber JAR file `flink-table-*.jar` is located in the `/lib` directory of a Flink release by default.
-* `flink-table-uber-blink`: Packages the API modules above plus the Blink specific modules into a distribution for most Table & SQL API use cases. The uber JAR file `flink-table-blink-*.jar` is located in the `/lib` directory of a Flink release by default.
+* `flink-table-common`: 公共模块，比如自定义函数、格式等需要依赖的。
+* `flink-table-api-java`: Table 和 SQL API，使用 Java 语言编写的，给纯 table 程序使用（还在早期开发阶段，不建议使用）
+* `flink-table-api-scala`: Table 和 SQL API，使用 Scala 语言编写的，给纯 table 程序使用（还在早期开发阶段，不建议使用）
+* `flink-table-api-java-bridge`: Table 和 SQL API，也支持 DataStream/DataSet API，给 Java 语言使用。
```
Comment (JingsongLi) [inline docs/dev/table/index.zh.md]: `Table 和 SQL API，也支持 DataStream/DataSet API`，好像不是这个意思，是Table/SQL API结合DataStream/DataSet一起用的用法。

File: docs/dev/table/index.zh.md [hunk 3]
```diff
@@ -25,41 +25,41 @@ specific language governing permissions and limitations
...
+* `flink-table-uber-blink`: 把上述模块以及 Blink planner 打包到一起，可以在大部分 Table & SQL API 场景下使用。打包到一起的 jar 文件 `flink-table-blink-*.jar` 默认会放到 Flink 发行版的 `/lib` 目录下。
 
-See the [common API](common.html) page for more information about how to switch between the old and new Blink planner in table programs.
+关于如何使用 old planner 以及 Blink planner，可以参考[公共 API](common.html)。 
 
-### Table Program Dependencies
+### Table 程序依赖
 
-Depending on the target programming language, you need to add the Java or Scala API to a project in order to use the Table API & SQL for defining pipelines:
+取决于你使用的编程语言，选择 Java 或者 Scala API 来构建你的程序：
```
Comment (JingsongLi) [inline docs/dev/table/index.zh.md]: 构建你的Table API和SQL的程序？

File: docs/dev/table/index.zh.md [hunk 4]
```diff
@@ -109,9 +108,9 @@ Internally, parts of the table ecosystem are implemented in Scala. Therefore, pl
 </dependency>
 {% endhighlight %}
 
-### Extension Dependencies
+### 扩展所需要的依赖
```
Comment (JingsongLi) [inline docs/dev/table/index.zh.md]: 扩展依赖？

**Comment by libenchao** [comment]
@JingsongLi Thanks for your review, I've addressed your comments.

**Comment by JingsongLi** [comment]
Thanks for updating, @libenchao merged.

**Pull request merged**